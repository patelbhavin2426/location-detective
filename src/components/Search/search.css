
html, body, #app {
    margin: 0;
    height: 100%;
    width: 100%;
    background: #fffcfc;
}

body {
    font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen-Sans,Ubuntu,Cantarell,Helvetica Neue,sans-serif;
}

ul {
    padding: 0;
}

li {
    list-style-type: none;
}

.loader {
    background-image: url("https://media.giphy.com/media/38msMCMSMUK7m/giphy.gif");
}

.container {
    max-width: 1200px;
    margin: 0 auto;
    padding: 50px;
}

.flex-center {
    display: flex;
    justify-content: center;
    align-items: center;
}

.btn-clear {
    border: none;
    background: transparent;
}

.nav-link {
    font-size: 18px;
    font-weight: bold;
    text-decoration: none;
    color: inherit;
}

.grid {
    display: flex;
    flex-wrap: wrap;
}

.search.space-around {
    background: white;
}


/* Search fields  */
form.field {
    padding: 20px 0 20px 0;
}

.space-wrapper {
    flex-basis: 40%;
}
input.location-query.form-control {
    flex-basis: 30%;
    margin-right: 10px;
}

input.form-control {
    height: 50px;
}

input.search-item.form-control {
    flex-basis: 70%;
}

button.btn.searchbox-submitbtn {
    border-radius: 8px;
}

.space-around {
    justify-content: space-around;
}

.header-lg {
    font-size: 35px;
    font-weight: 300;
    margin: 20px;
}

.avatar {
    width: 150px;
    height: 150px;
    border-radius: 3px;
    margin: 0 auto;
    display: block;
}

.center-text {
    text-align: center;
}

.link {
    color: rgb(187, 46, 31);
    text-decoration: none;
    font-weight: bold;
}


.card-list {
    margin: 20px 0;
    font-size: 20px;
}

.card-list li {
    display: flex;
    align-items: center;
    margin: 10px;
}

.card-list svg {
    margin-right: 10px;
}

.card-list a {
    font-weight: 500;
    color: inherit;
}


.bg-light {
    background: rgba(0, 0, 0, 0.08);
    border-radius: 3px;
}

input,
button,
.searchbox:before {
    padding: .35em .75em;
    border: none;
    font-size: 1.1rem;
    text-decoration: none;
}

/* LAYOUT */
.field {
    display: flex;
}

button.btn.searchbox-submitbtn {
    margin-left: 12px;
}
/* Medium devices (landscape tablets, 768px and up) */
@media (max-width: 500px) {

    .field {
        flex-direction: column;
    }

    input,
    button,
    .searchbox:before {
        padding: .35em .75em;
        border: none;
        font-size: 1em;
        text-decoration: none;
    }

    input.form-control {
        height: 150px;
    }

    .search .field {
        margin-left: 5%;
        margin-right: 5%;
    }

    .search input.form-control  {
        margin-top: 2%;
        margin-bottom: 2%;
    }

    button.btn.searchbox-submitbtn {
        margin-top: 10px;
        margin-left: 0px;
    }

}

.location-query {
    flex: 1 0 8em;
}

.searchbox-submitbtn {
    flex: 0 0 auto;
}

.search-field:before {
    flex: 0 0 2.75em;

    display: inline-flex;
    align-items: center;
    justify-content: center;
}

.location-query {
    background: #e9e9e9;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
}

.location-query:focus {
    background: white;
}

.searchbox-submitbtn {
    border-radius: 0 2px 2px 0;
    background: #9792E3;
    font-weight: 300;
    text-transform: uppercase;
    color: #EDFFEC;
}

p.inputError {
    text-align: center;
    color: #e45a5a;
    font-weight: 500;
}

/* Media Queries */
