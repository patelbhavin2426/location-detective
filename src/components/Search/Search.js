import React, {Component } from 'react'
import PropTypes from 'prop-types';
import { fetchPlaces } from '../../services'

import { Venues } from '../Venues'
import './search.css'

/* For Redux
* import { connect } from 'react-redux'
* import { actions } from '../../actions'
*/

class Search extends Component {

    constructor(props) {
        super(props)

        this.state = {
            'searchItem' : '',
            'zipCode':'',
            'venues':null,
            'error': null
        };

        this.searchVenues = this.searchVenues.bind(this);
        this.updateSearchItem = this.updateSearchItem.bind(this);
        this.updateZipCode = this.updateZipCode.bind(this);
    }

    searchVenues(event) {

        event.preventDefault();

        this.setState( {
            'venues':null,
            'error': null,
            'loading' : true
        });


        fetchPlaces(this.state.zipCode, this.state.searchItem)
            .then((data) => {

                this.setState({
                    'venues':data,
                    'error': null,
                    'loading' : false
                });

                // console.log(data);
                // This is for Redux
                // this.props.venuesReceived(data);

            })
            .catch((error) => {

                console.warn('Error fetching repos: ', error)

                this.setState({
                    venues : null,
                    error: 'There was an error fetching the locations.',
                    'loading' : false
                })
            });

    }

    updateZipCode (e) {

        this.setState({
            'zipCode' :e.target.value
        });
    }

    updateSearchItem (e) {

        this.setState({
            'searchItem' :e.target.value
        });
    }

    isLoading() {
        return this.state.error === null && this.state.venues === null;
    }

    render(){

        const {venues, error} = this.state;

        return (
            <React.Fragment>

                <div className="search space-around">
                     <form className="field" role="search" >
                        <div className="space-wrapper"></div>
                        <input onChange={this.updateZipCode} type="search" className="zipcode-query form-control" name='zipCode' value={this.state.zipCode} placeholder="City" />
                        <input onChange={this.updateSearchItem} type="search" className="search-item form-control" name='searchItem' value={this.state.searchItem} placeholder="Search Place" />
                        <button className="btn searchbox-submitbtn" onClick={this.searchVenues}>Search</button>
                        <div className="space-wrapper"></div>
                    </form>
                </div>

                <div className="place-container" >

                    { this.isLoading() && <div className="resultset" >
                        <div className={"loader"}></div>
                    </div> }

                    { error && <p>{error}</p> }

                    { venues && <Venues venues={venues} /> }
                </div>

            </React.Fragment>
        )
    }

}

/* For Redux
const stateToProps = (state) => {

    console.log("received ... ");

    return {
        venues : state.venues
    }
}

const dispatchToProps = (dispatch) => {

    console.log("sending.... ... ");

    return {
        venuesReceived: (venues) => {
            dispatch(actions.venuesReceived(venues));
            console.log("sent.. ... ");
        }
    }
}

export default connect(stateToProps, dispatchToProps)(Nav);

 */


export default Search;
