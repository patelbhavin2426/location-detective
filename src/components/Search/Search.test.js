import React from 'react';
import renderer from 'react-test-renderer';

import { shallow, mount } from "enzyme";

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


// For Redux
// import { Provider } from "react-redux";
// import configureMockStore from "redux-mock-store";


Enzyme.configure({ adapter: new Adapter() });
// const mockStore = configureMockStore();
// const store = mockStore({
//     venues : venues
// });

import Search from './Search'

describe("Search Component :: ", () => {

    it("should render without throwing an error", () => {
        expect(
            renderer.create(
                <Search/>
            )
        ).not.toBe(null);
    });


    it("should render search inputs UI", () => {

        // Act
        let component = renderer.create(
            <Search />
        ).toJSON();
        // snapshot
        expect(component).toMatchSnapshot();


        // Act
        component = mount(
            <Search />
        );
        expect((component).prop('state')).not.toEqual(null);

        // class check
        expect(component.find('.zipcode-query').text()).toEqual('');
        expect(component.find('.search-item').text()).toEqual('');
        expect(component.find('.searchbox-submitbtn').text()).toEqual('Search');


    });


    it("should change state with value change in input fields", () => {

        // Act
        let component = mount(
            <Search />
        );
        expect((component).prop('state')).not.toEqual(null);

        // basic check
        expect(component.find('.zipcode-query').text()).toEqual('');
        expect(component.find('.search-item').text()).toEqual('');
        expect(component.find('.searchbox-submitbtn').text()).toEqual('Search');

        // type in the text
        component.find('.zipcode-query').simulate('change', {target: {name: 'zipCode', value: 'Fremont'}});
        expect(component.state('zipCode')).toEqual('Fremont');

        component.find('.search-item').simulate('change', {target: {name: 'searchItem', value: 'food'}});
        expect(component.state('searchItem')).toEqual('food');

    });


    it("let's click on search button", () => {

        // Act
        let component = mount(
            <Search />
        );
        expect((component).prop('state')).not.toEqual(null);

        // basic check
        expect(component.find('.zipcode-query').text()).toEqual('');
        expect(component.find('.search-item').text()).toEqual('');
        expect(component.find('.searchbox-submitbtn').text()).toEqual('Search');

        // type in the text
        component.find('.zipcode-query').simulate('change', {target: {name: 'zipCode', value: 'Fremont'}});
        component.find('.search-item').simulate('change', {target: {name: 'searchItem', value: 'food'}});
        component.find('button').simulate('click');

        expect(component.state('loading')).toBe(true);

    });



});

