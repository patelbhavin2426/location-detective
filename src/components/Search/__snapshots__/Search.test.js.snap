// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component ::  should render search inputs UI 1`] = `
Array [
  <div
    className="search space-around"
  >
    <form
      className="field"
      role="search"
    >
      <div
        className="space-wrapper"
      />
      <input
        className="zipcode-query form-control"
        name="zipCode"
        onChange={[Function]}
        placeholder="City"
        type="search"
        value=""
      />
      <input
        className="search-item form-control"
        name="searchItem"
        onChange={[Function]}
        placeholder="Search Place"
        type="search"
        value=""
      />
      <button
        className="btn searchbox-submitbtn"
        onClick={[Function]}
      >
        Search
      </button>
      <div
        className="space-wrapper"
      />
    </form>
  </div>,
  <div
    className="place-container"
  >
    <div
      className="resultset"
    >
      <div
        className="loader"
      />
    </div>
  </div>,
]
`;
